ui/components/app/signature-request/signature-request.component.js

```diff
@@ -116,16 +124,30 @@ export default class SignatureRequest extends PureComponent {
             {this.context.t('sigRequest')}
           </div>
           <div className="signature-request-content__identicon-container">
-            <div className="signature-request-content__identicon-initial">
-              {domain.name && domain.name[0]}
-            </div>
-            <div className="signature-request-content__identicon-border" />
-            <Identicon address={fromAddress} diameter={70} />
+            {targetSubjectMetadata?.iconUrl ? (
+              <SiteIcon
+                icon={targetSubjectMetadata.iconUrl}
```

Comment:
@jpuri Where do you get the subjectMetadata from? Is it from the contract-metadata [here](https://github.com/MetaMask/contract-metadata)?

Reply:
I'm trying to see where I should I add OpenSea's logo for the contract id and any other associated info.

Reply:
It is populated by our inpage provider. You can see how the image is found here: https://github.com/MetaMask/providers/blob/main/src/siteMetadata.ts#L78

Reply:
So this is the favicon of the origin site.
This poses a security risk: If a malicious site adds OpenSea logo as their favicon, this would look legitimate upon signing.

Reply:
Is it possible to get this from the contract-metadata.json? May not be an ideal fit for OpenSea, but worth a try? 

Or could we just special case for OS for the short term. we can work together on a longer term fix.

Reply:
Circling back after conversations on slack. This looks good to us for the short term.